### PR TITLE
feat: display date as per user locale 

### LINKED ISSUE #353: Ensure consistent timestamp handling
* Should use the same format. 
* Assume UTC and convert to local timezone

### PR DESCRIPTION
**Pull request by PulkitBanta**
### Description
Timestamp values are formatted locally as per the user while being displayed.

### Fixes #353

### CODE REVIEW

File: src/components/Canary/CanaryStatusChart/index.js [hunk 1]
```diff
@@ -15,6 +14,7 @@ import { getCanaryGraph } from "../../../api/services/topology";
 import { Loading } from "../../Loading";
 import { useSearchParams } from "react-router-dom";
 import { getParamsFromURL } from "../utils";
+import { formatDate } from "../../../utils/formatDate";
```
Comment (moshloop) [inline src/components/Canary/CanaryStatusChart/index.js:17]: Can we extract all of the date-related functionality into utils/date.js ? any use of dayjs etc should be isolated to that file.

File: src/components/Hypothesis/EvidenceSection/index.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ const EvidenceAccordion: React.FC<{
             {title || <span className="text-gray-400">(no title)</span>}
             {date && (
               <div className="text-gray-400 text-sm">
-                {dayjs(date).format("HH:mm A, MMM DD, YYYY")}
+                {formatDate(date, "HH:mm A, MMM DD, YYYY")}
```
Comment (moshloop) [inline src/components/Hypothesis/EvidenceSection/index.tsx:68]: The data format should be in date.js -  different components shouldn't be using different formats.  it should be one:

* relative
* long (textual)
* timestamp 
* time

**Comment by moshloop** [comment]
I think incident details table also needs to be catered for

**Comment by moshloop** [comment]
I still don't see a change on incidents.

File: src/components/Canary/CanaryStatusChart/index.js [hunk 3]
```diff
@@ -55,11 +55,11 @@ export function CanaryStatusChart({ check, ...rest }) {
   }
 
   // @TODO: date should be formatted properly depending on selection, not just by DD/MM
-  let formatDate = (date) => dayjs(date).format("HH:mm");
+  let customFormatDate = (date) => formatDate(date, "HH:mm");
```
Comment (moshloop) [inline src/components/Canary/CanaryStatusChart/index.js:58]: formatDate should not accept a format - formats can only be defined in date.ts, and they must be consistently used. 
In this case relativeDate is probably better

File: src/components/Canary/CanaryStatusChart/index.js (hunk 1 above)
Comment (PulkitBanta) [reply src/components/Canary/CanaryStatusChart/index.js]: Sure, I will make that change

File: src/components/Canary/CanaryStatusChart/index.js (hunk 3 above)
Comment (PulkitBanta) [reply src/components/Canary/CanaryStatusChart/index.js:58]: I am defining a enum for the date format types which will be consistent and is going to have 3-4 types which will be used throughout the application

File: src/components/Canary/CanaryStatusChart/index.js [hunk 4]
```diff
@@ -15,6 +14,7 @@ import { getCanaryGraph } from "../../../api/services/topology";
 import { Loading } from "../../Loading";
 import { useSearchParams } from "react-router-dom";
 import { getParamsFromURL } from "../utils";
+import { DATE_FORMATS, formatDate } from "../../../utils/date";
 
 // @TODO: duration should be formatted properly, not just by ms
 const formatDuration = (duration) => `${duration}ms`;
```
Comment (moshloop) [inline src/components/Canary/CanaryStatusChart/index.js:20]: this formatDuration method itself should move, not just the formats

**Comment by Prashanth-Nelli** [comment]
closing this PR as the requirement is covered in https://github.com/flanksource/flanksource-ui/pull/559

**Pull request closed without merging**